plugins/woocommerce-blocks/assets/js/data/cart/thunks.ts

```diff
@@ -207,8 +209,12 @@ export const addItemToCart =
 				path: `/wc/store/v1/cart/add-item`,
 				method: 'POST',
 				data: {
-					id: productId,
-					quantity,
+					...additionalData,
+					...{
+						id: productId,
+						quantity,
+						variation
+					},
```

Comment:
You don't need to add the `id` and the `quantity` into a separate object and then spreading it again, this should suffice:
```suggestion
					...additionalData,
				    id: productId,
					quantity,
					variation,
```